autopilot/include/autopilot.h

```diff
@@ -0,0 +1,191 @@
+#include <boost/thread/mutex.hpp>
+#include <sstream>
+
+#include "fin_control/EnableReportAngles.h"
+#include "fin_control/ReportAngle.h"
+#include "fin_control/SetAngles.h"
+#include "jaus_ros_bridge/ActivateManualControl.h"
+#include "mission_manager/AttitudeServo.h"
+#include "mission_manager/DepthHeading.h"
+#include "mission_manager/FixedRudder.h"
+#include "mission_manager/ReportExecuteMissionState.h"
+#include "mission_manager/ReportHeartbeat.h"
+#include "mission_manager/SetBehavior.h"
+#include "pid.h"
+#include "sensor_msgs/Imu.h"
+#include "thruster_control/ReportRPM.h"
+#include "thruster_control/SetRPM.h"
+
+#define NODE_VERSION "2.02x"
+
+class AutoPilotNode {
+ public:
+  AutoPilotNode(ros::NodeHandle& node_handle);
+  void Start();
+  void Stop();
+
+ private:
+  ros::NodeHandle nh;
+  ros::Publisher auto_pilot_in_control_pub;
+  ros::Publisher thruster_pub;
+  ros::Publisher fins_control_pub;
+  ros::Publisher fins_enable_reporting_pub;
+
+  ros::Subscriber jaus_ros_sub;
+  ros::Subscriber thruster_sub;
+
+  ros::Subscriber desiredRoll_sub_;
+  ros::Subscriber desiredYaw_sub_;
+  ros::Subscriber desiredPitch_sub_;
+
+  ros::Subscriber correctedData_sub_;
+  ros::Subscriber missionStatus_sub_;
+  ros::Timer missionMgrHeartbeatTimer;
+
+  ros::Subscriber missionMgrHeartBeat_sub_;
+  ros::Subscriber setAttitudeBehavior_sub_;
+  ros::Subscriber setDepthHeadingBehavior_sub_;
+  ros::Subscriber setFixedRudderBehavior_sub_;
+
+  control_toolbox::Pid roll_pid_controller;
+  control_toolbox::Pid pitch_pid_controller;
+  control_toolbox::Pid yaw_pid_controller;
+  control_toolbox::Pid depth_pid_controller;
+
+  bool auto_pilot_in_control;
+  bool mission_mode;
+
+  double roll_pgain;
+  double roll_igain;
+  double roll_dgain;
+  double roll_imax;
+  double roll_imin;
+
+  double pitch_pgain;
+  double pitch_igain;
+  double pitch_dgain;
+  double pitch_imax;
+  double pitch_imin;
+
+  double yaw_pgain;
+  double yaw_igain;
+  double yaw_dgain;
+  double yaw_imax;
+  double yaw_imin;
+
+  double depth_pgain;
+  double depth_igain;
+  double depth_dgain;
+  double depth_imax;
+  double depth_imin;
+
+  void correctedDataCallback(const pose_estimator::CorrectedData& data);
+  void missionStatusCallback(const mission_manager::ReportExecuteMissionState& data);
+
+  void HandleActivateManualControl(const jaus_ros_bridge::ActivateManualControl& data);
+
+  void mixactuators(double roll, double pitch, double yaw);
+  double radiansToDegrees(double radians);
+  double degreesToRadians(double degrees);
+  void missionMgrHeartbeatTimeout(const ros::TimerEvent& timer);
+
+  void workerFunc();
+
+  boost::mutex m_mutex;
+
+  double currentRoll;
+  double currentPitch;
+  double currentYaw;
+  double currentDepth;
+
+  double desiredRoll;
+  double desiredPitch;
+  double desiredYaw;
+  double desiredDepth;
+  double desiredRudder;
+  double desiredSpeed;
+
+  double maxAllowedThrusterRpm;
+  double minimalspeed;  // knots
```

Comment:
@patapufate I see a mixture of camel case and snake case variable naming. Should we normalize?

Reply:
@ChrisScianna do you have any preference on this?